app/src/main/java/org/simple/clinic/drugs/selection/custom/CustomDrugEntryModel.kt

```diff
@@ -0,0 +1,38 @@
+package org.simple.clinic.drugs.selection.custom
+
+import org.simple.clinic.drugs.search.DrugFrequency
+
+data class CustomDrugEntryModel(
+    val openAs: OpenAs,
+    val drugName: String?,
+    val dosage: String?,
+    val dosageHasFocus: Boolean?,
+    val frequency: DrugFrequency?
```

Comment:
I am quite confused here. `OpenAs` tracks a `Drug` when opened via the `FromDrugList` parameter. It can also track the `drugName` separately when opened via the `FromDrugName` parameter.

At the same time, `CustomDrugEntryModel` also tracks the `drugName`, `dosage`, and `frequency`, which are also present in `Drug` (when opened via the `FromDrugList`).

What is the purpose of tracking these multiple, similar pieces of information in the model here?

Reply:
> Because when in CREATE mode we open the bottom sheet from the DrugSearchScreen where two cases are possible..
> 
> 1. The user has typed a drug name which is not part of the drug list.
> 2. The user has typed a drug and finds in the drug list and then clicks on it.

So, OpenAs.New.FromDrugList is the first use case where we will load the `drugName`, `dosage`, `frequency` from the drug object. 
Whereas in the second use case is `OpenAs.New.FromDrugName` we only have a drug name and we are going to set up the bottom sheet accordingly. In this case we will not have any value for drug dosage and the value for frequency will be null as well. 

So when the bottom sheet is opened according to the OpenAs value the sheet is initialised. Although, then the initial values get saved in the model and these values can be further edited by the user..

Reply:
Ah, understood. One question then, do we _need_ to track the passed in drug name or `Drug` object in the model at all? Or are we using it only for prefilling the initial details, in which case, this can be handled in the factory method without needing to track it in the model?

Reply:
I have already removed the drug object from the model.. Yes, the `drugName` parameter is only updated once in the model when it's initialised but it's used multiple times to set the title text. Which includes the `drugName`, `dosage`, `frequency`. Which is why I have added in the model.

Reply:
The title text gets updated each time the user edits the `dosage` or the `frequency`

Reply:
Hmm, ok. I still don't understand completely but we can merge this in now and take a look when the feature is completed.